File: touralbum/app/src/main/java/com/example/touralbum/ui/reminder/receiver/ClockReceiver.java
package com.example.touralbum.ui.reminder.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.example.touralbum.ui.reminder.activity.ClockActivity;
import com.example.touralbum.ui.reminder.dao.EventDao;
import com.example.touralbum.ui.reminder.entity.Event;
import com.example.touralbum.ui.reminder.util.WakeLockUtil;

public class ClockReceiver extends BroadcastReceiver {
    private static final String TAG = "ClockReceiver";
    public static final String EXTRA_EVENT_ID = "extra.event.id";
    public static final String EXTRA_EVENT_REMIND_TIME = "extra.event.remind.time";
    public static final String EXTRA_EVENT = "extra.event";
    private EventDao mEventDao = EventDao.getInstance();

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG, "onReceive: " + intent.getAction());
        WakeLockUtil.wakeUpAndUnlock();
        postToClockActivity(context, intent);
    }

    private void postToClockActivity(Context context, Intent intent) {
        Intent i = new Intent();
        i.setClass(context, ClockActivity.class);
        i.putExtra(EXTRA_EVENT_ID, intent.getIntExtra(EXTRA_EVENT_ID, -1));
        Event event = mEventDao.findById(intent.getIntExtra(EXTRA_EVENT_ID, -1));
        if (event == null) {
            return;
        }
        i.putExtra(EXTRA_EVENT_REMIND_TIME, intent.getStringExtra(EXTRA_EVENT_REMIND_TIME));
        i.putExtra(EXTRA_EVENT, event);
        i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        context.startActivity(i);
    }

    public ClockReceiver() {
        super();
        Log.d(TAG, "ClockReceiver: Constructor");
    }
}

File: touralbum/app/src/main/java/com/example/touralbum/ui/reminder/receiver/ClockReceiver.kt
package com.example.touralbum.ui.reminder.receiver

import android.content.*
import android.util.Log
import com.example.touralbum.ui.reminder.activity.ClockActivity
import com.example.touralbum.ui.reminder.dao.EventDao
import com.example.touralbum.ui.reminder.util.WakeLockUtil

class ClockReceiver : BroadcastReceiver() {
    private val mEventDao: EventDao = EventDao.instance
    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive: " + intent.action)
        WakeLockUtil.wakeUpAndUnlock()
        postToClockActivity(context, intent)
    }

    private fun postToClockActivity(context: Context, intent: Intent) {
        val i = Intent()
        i.setClass(context, ClockActivity::class.java)
        i.putExtra(EXTRA_EVENT_ID, intent.getIntExtra(EXTRA_EVENT_ID, -1))
        val event = mEventDao.findById(intent.getIntExtra(EXTRA_EVENT_ID, -1))
            ?: return
        i.putExtra(EXTRA_EVENT_REMIND_TIME, intent.getStringExtra(EXTRA_EVENT_REMIND_TIME))
        i.putExtra(EXTRA_EVENT, event)
        i.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        context.startActivity(i)
    }

    companion object {
        private const val TAG = "ClockReceiver"
        const val EXTRA_EVENT_ID = "extra.event.id"
        const val EXTRA_EVENT_REMIND_TIME = "extra.event.remind.time"
        const val EXTRA_EVENT = "extra.event"
    }

    init {
        Log.d(TAG, "ClockReceiver: Constructor")
    }
}